streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreFactory.java

```diff
@@ -75,4 +76,78 @@ default void configure(final StreamsConfig config) {
 
     boolean isCompatibleWith(StoreFactory storeFactory);
 
+    class FactoryWrappingStoreBuilder<T extends StateStore> implements StoreBuilder<T> {
+
+        private final StoreFactory storeFactory;
+
+        public FactoryWrappingStoreBuilder(final StoreFactory storeFactory) {
+            this.storeFactory = storeFactory;
+        }
+
+        public StoreFactory storeFactory() {
+            return storeFactory;
+        }
+
+        @Override
+        public boolean equals(final Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+
+            final FactoryWrappingStoreBuilder<?> that = (FactoryWrappingStoreBuilder<?>) o;
+
+            return storeFactory.isCompatibleWith(that.storeFactory);
+        }
+
+        @Override
+        public int hashCode() {
+            return storeFactory.hashCode();
+        }
+
+        @Override
+        public StoreBuilder<T> withCachingEnabled() {
+            throw new IllegalStateException("Should not try to modify StoreBuilder wrapper");
+        }
+
+        @Override
+        public StoreBuilder<T> withCachingDisabled() {
+            throw new IllegalStateException("Should not try to modify StoreBuilder wrapper");
```

Comment:
I'm not 100% sure it's necessary, but I think we might as well delegate to the StoryFactory here as well. If the StoryFactory has/needs this method for whatever reason then it seems theoretically possible for it to be callled during the topology building process (whereas imo it's highly unlikely for withLoggingEnabled or withCachingEnabled to be called on this, especially if those methods don't even  exist on StoryFactory)